ChangeLog.md

```diff
@@ -1,3 +1,18 @@
+### 0.4.0
+
+#### chruby.sh
+
+* Use the installed Ruby directory name for setting `GEM_HOME`. (@eregon)  
+  This guarantees a unique `GEM_HOME` per installed Ruby, even if they have
+  the same `RUBY_VERSION`.  This is important for native extensions, which
+  might compile differently based on build-time flags such as `--enable-shared`.
```

Comment:
We already confirmed that RubyGems keeps shared vs non-shared extensions in separate directories.

Reply:
Yes, but that's impractical and I think nobody wants the behavior of "half-installed gems" (i.e., the gem is there but the extension isn't compiled because of different build-time flags, example warning: `Ignoring msgpack-1.3.0 because its extensions are not built. Try: gem pristine msgpack --version 1.3.0`).
I'm also quite confident we could find a set of build-time flags that would make extensions incompatible while they would use the same directory.

I can drop the sentence in the ChangeLog though if you think it's confusing.